worlds/apquest/test/test_easy_mode.py

```diff
@@ -0,0 +1,113 @@
+            # Manually checking the dependency in the previous function was a bit of a hassle, wasn't it?
+            # Now we are checking four locations. It would be even longer as a result.
+            # Well, there is another option. It's the assertAccessDependency function of WorldTestBase.
+            self.assertAccessDependency(
+                [
+                    "Bottom Right Room Right Chest",
+                    "Bottom Right Room Left Chest",
+                    "Right Room Enemy Drop",
+                    "Final Boss Defeated",  # Reminder: APQuest's victory condition uses this event location
+                ],
+                [["Sword"]],
+            )
+
+            # The assertAccessDependency function is a bit complicated, so let's discuss what it does.
+            # By default, the locations argument must contain *every* location that *hard-depends* on the items.
+            # So, in our case: If every item except Sword is collected, *exactly* these four locations are unreachable.
+
+            # The possible_items argument is initially more intuitive, but has some complexity as well.
+            # In our case, we only care about one item. But sometimes, we care about multiple items at once.
+            # This is why we pass a list of lists. We'll discuss this more when we test hard mode logic.
+
+        # Let's do one more test: That the key is required for the Button.
+        with self.subTest("Test that the Key is required to activate the Button"):
+            # The Button is not the only thing that depends on the Key.
+            # As explained above, the locations list must be exhaustive.
+            # Thus, we would have to add the "Top Left Room Chest" as well.
+            # However, we can set "only_check_listed" if we only want the Top Left Room Button location to be checked.
+            self.assertAccessDependency(
+                ["Top Left Room Button"],
+                [["Key"]],
+                only_check_listed=True,
+            )
+
+    def test_easy_mode_health_upgrades(self) -> None:
+        # For our second test, let's make sure that we have two Health Upgrades with the correct classification.
+
+        # We can find the Health Upgrades in the itempool like this:
+        health_upgrades = self.get_items_by_name("Health Upgrade")
+
+        # First, let's verify there's two of them.
+        with self.subTest("Test that there are two Health Upgrades in the pool"):
+            self.assertEqual(len(health_upgrades), 2)
+
+        # Then, let's verify that they have the useful classification and NOT the progression classification.
+        with self.subTest("Test that the Health Upgrades in the pool are useful, but not progression."):
+            self.assertTrue(all(health_upgrade.useful for health_upgrade in health_upgrades))
+            self.assertFalse(any(health_upgrade.advancement for health_upgrade in health_upgrades))
```

Comment:
should we mention that "advancement" is a funny name for "progression"? (in fact, can we alias that so world devs don't need to learn it?)